### PR TITLE
Logging by step

### PR DESCRIPTION
**Pull request by fhtanaka**
Now the simulation logs its data during the simulation instead of only doing it at the end. It will do the extraction according to the configuration attribute `reportInterval`. 
Please note that this interval is counted separately from the `stepCount`; by doing this, these values don't need to be multiples of each other.

I also added a fix for the window size. Its value was absolute before now I added it on the config to be proportional to the screen size.

### CODE REVIEW

File: lib/Simulation/Job.py [hunk 1]
```diff
@@ -21,7 +21,7 @@ def __init__(self,jobClass):
         self.startHour =  random.randint(jobClass.minStartHour,jobClass.maxStartHour)
         indexes = np.where(jobClass.workDays)[0]
         if len(indexes) < self.activityPerWeek+1:
-            print("warning, activities per week is lower than the optional workdays")
```
Comment (caranha) [inline lib/Simulation/Job.py:24]: TODO: Add an issue to move "prints" to a logging framework.

File: lib/Simulation/Simulator.py [hunk 2]
```diff
@@ -12,9 +12,55 @@
...
+    'symptomaticMinutes',
+    'severeTimeStamp',
+    'severeDay',
+    'severeHour',
+    'severeMinutes',
+]
+
 class Simulator:
-    def __init__(self, osmMap, jobCSVPath, agentNum = 1000, threadNumber = 4, infectedAgent = 5, vaccinationPercentage = 0.0):
+    def __init__(self, osmMap, jobCSVPath, agentNum = 1000, threadNumber = 4, infectedAgent = 5, vaccinationPercentage = 0.0, reportPath="report/", reportInterval=10):
```
Comment (caranha) [inline lib/Simulation/Simulator.py:63]: Change "report/" to "report"

File: lib/Simulation/Simulator.py [hunk 3]
```diff
@@ -180,17 +236,22 @@ def step(self,steps = 3600):
...
         print("Finished finalizing the infection")
-        self.stepCount += steps
+        self.stepCount += stepSize
         self.summarize()
         self.printInfectionLocation()
+
+        if self.reportIntervalCount % self.reportInterval == 0:
+            print("Extracting data")
+            self.extract()
+        self.reportIntervalCount += 1
```
Comment (caranha) [inline lib/Simulation/Simulator.py:254]: Change to a "cooldown" pattern:
```python
        if self.reportIntervalCount < 1:
            print("Extracting data")
            self.extract()
            self.reportIntervalCount = self.reportInterval
        self.reportIntervalCount -= 1
```

File: lib/Simulation/Simulator.py [hunk 4]
```diff
@@ -236,41 +297,35 @@ def summarize(self):
...
 
-        with open(join(path,'infection_summary.csv'), 'w', newline='') as csvfile:
-            fieldnames = ['Day','Hour','Minutes','CurrentStep','Susceptible','Exposed','Infectious','Recovered']
-            writer = csv.DictWriter(csvfile, fieldnames=fieldnames)
+        summaryFilePath = join(self.reportPath,'infection_summary.csv')
+        with open(summaryFilePath, 'w', newline='') as summaryFile:
+            writer = csv.DictWriter(summaryFile, fieldnames=summaryFieldnames)
             writer.writeheader()
             for x in self.infectionHistory:
                 writer.writerow(x)
```
Comment (caranha) [inline lib/Simulation/Simulator.py:308]: TODO: Remember where in the "infection history" we stopped saving the log, and continue later.

Possible way to do this even better:
- every time we add a line to "infection history", we copy the dict to a variable named "logbuffer". 
- At "extract", we write "logbuffer" to the log, and clear the log

File: lib/Simulation/Simulator.py [hunk 5]
```diff
@@ -236,41 +297,35 @@ def summarize(self):
...
          
-        with open(join(path,'agents.csv'), 'w', newline='') as csvfile:
-            fieldnames = getAgentKeys()
-            writer = csv.DictWriter(csvfile, fieldnames=fieldnames)
+        agentsFilePath = join(self.reportPath,'agents.csv')
+        with open(agentsFilePath, 'w', newline='') as agentsFile:
+            agentFieldnames = getAgentKeys()
+            writer = csv.DictWriter(agentsFile, fieldnames=agentFieldnames)
             writer.writeheader()
             for x in self.agents:
```
Comment (caranha) [inline lib/Simulation/Simulator.py:326]: TO Issue: Consider if we want to keep the history of the agents in the future (maybe this is too much data)

**Review by caranha — APPROVED** [review]

**Review by rithgroove — APPROVED** [review]

**Pull request merged**